lib/vm/src/memory.rs

```diff
@@ -497,12 +550,17 @@ impl VMMemory {
     /// are natively supported
     /// - VMOwnedMemory -> VMMemory
     /// - Box<dyn LinearMemory + 'static> -> VMMemory
-    pub fn from_custom<IntoVMMemory>(memory: IntoVMMemory) -> VMMemory
+    pub fn from_custom<IntoVMMemory>(memory: IntoVMMemory) -> Self
     where
-        IntoVMMemory: Into<VMMemory>,
+        IntoVMMemory: Into<Self>,
     {
         memory.into()
     }
+
+    /// Copies this memory to a new memory
+    pub fn fork(&mut self) -> Result<Box<dyn LinearMemory + 'static>, MemoryError> {
```

Comment:
Let's use clone. I think is a more accurate naming

Reply:
Renamed to `duplicate()`.